File: ICHI_Client/src/app/dtos/trxtransaction.dto.ts
import {
  IsString,
  IsNotEmpty,
  IsPhoneNumber,
  IsBoolean,
  IsNumber,
} from 'class-validator';
import { MasterEntityDTO } from './master.entity.dto';
import { ProductImage } from '../models/product.image';
import { CartModel } from '../models/cart.model';
import { CustomerModel } from '../models/customer.model';

export class TrxTransactionDTO {
  trxTransactionId: number;
  customerId: string;
  employeeId: string;
  fullName: string;
  phoneNumber: string;
  address: string;
  amount: number;
  paymentTypes: string;
  checkOrder: boolean;
  carts: any;
  constructor(
    trxTransactionId: number,
    customerId: string,
    employeeId: string,
    fullName: string,
    phoneNumber: string,
    address: string,
    amount: number,
    paymentTypes: string,
    checkOrder: boolean,
    carts: any
  ) {
    this.trxTransactionId = trxTransactionId;
    this.customerId = customerId;
    this.employeeId = employeeId;
    this.fullName = fullName;
    this.phoneNumber = phoneNumber;
    this.address = address;
    this.amount = amount;
    this.paymentTypes = paymentTypes;
    this.checkOrder = checkOrder;
    this.carts = carts;
  }
}

File: ICHI_API/Model/TrxTransactionDTO.cs
using ICHI_CORE.Domain.MasterModel;

namespace ICHI_API.Model
{
  public class TrxTransactionDTO
  {
    public int? TrxTransactionId { get; set; }

    public string CustomerId { get; set; }

    public string? EmployeeId { get; set; }

    public string? OrderStatus { get; set; }

    public string? FullName { get; set; }

    public string? PhoneNumber { get; set; }

    public string? Address { get; set; }

    public string PaymentTypes { get; set; }

    public List<Cart> Carts { get; set; }

    public decimal? Amount { get; set; }

    public bool? CheckOrder { get; set; }
  }
}
